kratos/utilities/math_utils.h

```diff
@@ -738,37 +676,77 @@ class MathUtils
      * @param rA Is the input matrix
      * @return The determinant of the 4x4 matrix
      */
-    static inline TDataType Det4(const MatrixType& rA)
+    template<class TMatrixType>
+    static inline TDataType Det4(const TMatrixType& rA)
```

Comment:
Wondering if makes sense to make `Det2`, `Det3` and `Det4` private to force a unique `Det` and `GeneralizedDet` interface.

Reply:
Just thinking, rather than making `Det2`, `Det3`, `Det4` private, what if we template it and keep it public, and make the `GeneralizedDet` interface using those. So if templated elements need it, then templated versions can be used. Usually the matrix size is templated in `elements` ryt?

Reply:
The matrix size is templated in the fluid elements, but I don't think this is the case in all the elements. In any case, this would imply going through all the uses of `Det` in Kratos to add the template argument. I'd prefer to stay without the template, but if we decide to add it, the changes are IMO out of the scope of this PR.

Reply:
End of the story then. Thanks for clarifying.